ospfd/ospf_ri.c

```diff
@@ -2047,13 +2102,994 @@ DEFUN (no_pce_cap_flag,
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ip_ospf_router_info,
-       show_ip_ospf_router_info_cmd,
-       "show ip ospf router-info",
-       SHOW_STR
-       IP_STR
-       OSPF_STR
-       "Router Information\n")
+DEFPY(flex_algo_defn, flex_algo_defn_cmd, "flexible-algorithm (128-255)",
+      "Define a Flexible Algorithm Definition\n"
+      "Assign a unique number to the Flexible Algorithm\n")
+{
+	int id;
+	struct flex_algo_defn *fad;
+	bool new_fad = false;
+
+	if (!ospf_ri_enabled(vty))
+		return CMD_WARNING_CONFIG_FAILED;
+
+	id = atoi(argv[1]->arg);
+	if (id < 0 || id > FLEX_ALGO_ID_MAX)
+		return CMD_ERR_NO_MATCH;
+
+	fad = flex_algo_lookup_defn(OspfRI.fad_info.fads, (uint8_t)id, true,
+				    &new_fad);
+	assert(fad);
+	if (new_fad) {
+		OspfRI.fad_info.num_fads++;
+		/* Refresh RI LSA if already engaged */
+		ospf_router_info_schedule(REFRESH_THIS_LSA);
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFPY(no_flex_algo_defn, no_flex_algo_defn_cmd,
+      "no flexible-algorithm (128-255)",
+      NO_STR
+      "Specify a Flexible Algorithm Definition\n"
+      "Unique number assigned to the Flexible Algorithm\n")
+{
+	int id;
+	struct flex_algo_defn *fad;
+
+	id = atoi(argv[2]->arg);
+	if (id < 0 || id > FLEX_ALGO_ID_MAX)
+		return CMD_ERR_NO_MATCH;
```

Comment:
same as above. Please take care of this for all the new clis.

Reply:
Thanks once again 👍